index.js

```diff
@@ -29,6 +58,25 @@ module.exports.captureExit = function() {
   exit = process.exit;
 
   process.exit = function(code) {
+    if (handlers.length === 0 && lastTime === undefined) {
+      // synchronously exit.
+      //
+      // We do this brecause either
+      //
+      //  1.  The process exited due to a call to `process.exit` but we have no
+      //      async work to do because no handlers had been attached.  It
+      //      doesn't really matter whether we take this branch or not in this
+      //      case.
+      //
+      //  2.  The process exited naturally.  We did our async work during
+      //      `beforeExit` and are in this function because someone else has
+      //      called `process.exit` during an `on('exit')` hook.  The only way
+      //      for us to preserve the exit code in this case is to exit
+      //      synchronously.
+      //
+      exit.call(process, code);
```

Comment:
if we have no handlers and no lastTime, should we also `return` at the end of this branch  as  I believe the flush bellow to no longer be required.